### ISSUE TITLE
Make the process of `calculateRequirementsForAssignments` parallel.

### ISSUE THREAD

**Issue by denkensk** [kubernetes-sigs/kueue#16]
This can be done in parallel

https://github.com/kubernetes-sigs/kueue/blob/f3b25fd960ec04eaa973cc318965e0d50d36b3f8/pkg/scheduler/scheduler.go#L74-L76

We should check what else can be parallelized and set the number of threads via configuration.

/kind feature

**Comment by alculquicondor** [comment]
This should be simple enough

/help

**Comment by kannon92** [comment]
Hello!  I'm interested in pulling this issue.  

I think the code changed a bit since this was posted.  

Is https://github.com/kubernetes-sigs/kueue/blob/6eeb73b1c6b9338f101ab095a39b1434a2f926ba/pkg/scheduler/scheduler.go#L185 this the code that we want to run in parallel now?

**Comment by ahg-g** [comment]
Thanks Kevin and welcome to Kueue :)

Until we create scalability tests to understand where the bottlenecks are, this issues may not be a priority.

But another issue that needs some attention is https://github.com/kubernetes-sigs/kueue/issues/259; would you be interested in that one?

**Comment by kannon92** [comment]
Yea, that sounds great!  And thank you!  

I've been attending the calls for the working group so I'm interested in helping out where I can.

**Comment by ahg-g** [comment]
Sounds good, feel free to assign it to yourself, for some reason I can't.

**Comment by alculquicondor** [comment]
@kannon92 any progress on this?

**Comment by kannon92** [comment]
I have made 0 progress on this.

**Comment by alculquicondor** [comment]
Should we leave it open for other contributors or can you still handle it?

**Comment by kannon92** [comment]
Yea, anyone who is interested should feel free to take it.

**Comment by alculquicondor** [comment]
/close

So far the biggest bottlenecks in Kueue have been the API calls, which @gabesaba optimized.

I don't think we need to introduce parallel computations yet.
